suite-common/wallet-utils/src/transactionUtils.ts

```diff
@@ -46,30 +46,47 @@ export const parseDateKey = (key: string) => {
     return d;
 };
 
+/* Convert date to string in YYYY-MM format */
+export const getMonthKey = (d: Date) => `${d.getFullYear()}-${d.getMonth() + 1}`;
+
+/** Parse Date object from a string in YYYY-MM format */
+export const parseMonthKey = (key: string) => {
+    const parts = key.split('-');
+    const d = new Date(Number(parts[0]), Number(parts[1]) - 1);
+    return d;
+};
+
 /**
  * Returns object with transactions grouped by a date. Key is a string in YYYY-MM-DD format.
  * Pending txs are assigned to key 'pending'.
  *
  * @param {WalletAccountTransaction[]} transactions
  */
-export const groupTransactionsByDate = (transactions: WalletAccountTransaction[]) =>
+export const groupTransactionsByDate = (
+    transactions: WalletAccountTransaction[],
+    groupSize: 'day' | 'month' = 'day',
+) => {
     // Note: We should use ts-belt for sorting this array but currently, there can be undefined inside
     // Built-in sort doesn't include undefined elements but ts-belt does so there will be some refactoring involved.
-    [...transactions]
-        // There could be some undefined/null in array, nut sure how it happens. Maybe related to pagination?
-        .filter(transaction => !!transaction)
-        .sort(sortByBlockHeight)
-        .reduce<{ [key: string]: WalletAccountTransaction[] }>((r, item) => {
-            const key =
-                item.blockHeight && item.blockHeight > 0 && item.blockTime && item.blockTime > 0
-                    ? getDateKey(new Date(item.blockTime * 1000))
-                    : 'pending';
-            const prev = r[key] ?? [];
-            return {
-                ...r,
-                [key]: [...prev, item],
-            };
-        }, {});
+    const keyFormatter = groupSize === 'day' ? getDateKey : getMonthKey;
+    return (
+        [...transactions]
+            // There could be some undefined/null in array, nut sure how it happens. Maybe related to pagination?
+            .filter(transaction => !!transaction)
+            .sort(sortByBlockHeight)
+            .reduce<{ [key: string]: WalletAccountTransaction[] }>((r, item) => {
+                const key =
+                    item.blockHeight && item.blockHeight > 0 && item.blockTime && item.blockTime > 0
+                        ? keyFormatter(new Date(item.blockTime * 1000))
+                        : 'pending';
+                const prev = r[key] ?? [];
+                return {
+                    ...r,
+                    [key]: [...prev, item],
+                };
+            }, {})
+    );
+};
```

Comment:
This will break desktop suite. Check usages of this function. You should move formatting level up to only affect mobile app